Source/ModelGroups/Mesh_PrimitiveMode.cs

```diff
@@ -1,488 +1,356 @@
-                                break;
-                            }
-                        case Propertyname.Mode_Triangle_Strip:
-                            {
-                                indices = properties.Find(e => e.name == Propertyname.IndicesValues_TriangleStrip);
-                                break;
-                            }
-                        case Propertyname.Mode_Triangle_Fan:
-                            {
-                                indices = properties.Find(e => e.name == Propertyname.IndicesValues_TriangleFan);
-                                break;
-                            }
-                        case Propertyname.Mode_Triangles:
-                            {
-                                indices = properties.Find(e => e.name == Propertyname.IndicesValues_Triangles);
-                                break;
-                            }
-                        case Propertyname.IndicesComponentType_None:
-                            {
-                                indices = null;
-                                break;
-                            }
-                    }
-                    wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].Indices = indices.value;
+                        new Vector3(-0.5f,-0.5f, 0.0f),
+                        new Vector3( 0.5f,-0.5f, 0.0f),
+                        new Vector3( 0.5f, 0.5f, 0.0f),
+                        new Vector3(-0.5f,-0.5f, 0.0f),
+                        new Vector3( 0.5f, 0.5f, 0.0f),
+                        new Vector3(-0.5f, 0.5f, 0.0f),
+                    };
                 }
-                else if (property.name == Propertyname.IndicesComponentType_Byte ||
-                         property.name == Propertyname.IndicesComponentType_Short ||
-                         property.name == Propertyname.IndicesComponentType_Int)
+                meshPrimitive.Mode = Runtime.MeshPrimitive.ModeEnum.TRIANGLES;
+                properties.Add(new Property(PropertyName.Mode, meshPrimitive.Mode));
+            }
+
+            void SetIndicesPoints(List<Property> properties, Runtime.MeshPrimitive meshPrimitive)
+            {
+                List<int> pointsIndices = new List<int>();
+                for (int x = 0; x <= 1023; x++)
                 {
-                    wrapper.Scenes[0].Nodes[0].Mesh.MeshPrimitives[0].IndexComponentType = property.value;
+                    pointsIndices.Add(x);
                 }
-                else if (property.name == Propertyname.IndicesComponentType_None)
+                meshPrimitive.Indices = pointsIndices;
+                properties.Add(new Property(PropertyName.IndicesValues, "[0 - 1023]"));
+            }
+
+            void SetIndicesLines(List<Property> properties, Runtime.MeshPrimitive meshPrimitive)
+            {
+                Runtime.MeshPrimitive defaultMeshPrimitive = MeshPrimitive.CreateSinglePlane();
```

Comment:
This is really weird. I think you should probably factor out the creation of the positions for a single plane into its own function.

Reply:
Fixed